### ISSUE TITLE
Only some files are restored with restore stack

### ISSUE THREAD

**Issue by koterpillar** [fpco/cache-s3#4]
I have successfully saved a cache file using `cache-s3 save stack`. Now I'm restoring it, and only some files are restored:

```
ls -l /home/xxxxxxxxxxxxxx/.stack/
total 8
drwxrwxr-x 3 501 501 4096 Jan 29 22:30 precompiled
drwxrwxr-x 3 501 501 4096 Jan 29 22:52 programs
```

I have added logging of each restored file into `restoreFile'` like this:

```haskell
    restoreFile' fi = do
      liftIO $ print fi
      -- original function
```

This is the _end_ of the output of `cache-s3 restore stack`:

```
FileInfo {filePath = "/home/xxxxxxxxxxxxxx/.stack/precompiled", fileUserId = 501, fileUserName = "xxxxxxxxxxxxxx", fileGroupId = 501, fileGroupName = "xxxxxxxxxxxxxx", fileMode = 509, fileSize = 0, fileType = FTDirectory, fileModTime = 1517265003}
FileInfo {filePath = "/home/xxxxxxxxxxxxxx/.stack/precompiled/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743", fileUserId = 501, fileUserName = "xxxxxxxxxxxxxx", fileGroupId = 501, fileGroupName = "xxxxxxxxxxxxxx", fileMode = 509, fileSize = 0, fileType = FTDirectory, fileModTime = 1517265003}
FileInfo {filePath = "/home/xxxxxxxxxxxxxx/.stack/precompiled/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/ghc-8.2.2", fileUserId = 501, fileUserName = "xxxxxxxxxxxxxx", fileGroupId = 501, fileGroupName = "xxxxxxxxxxxxxx", fileMode = 509, fileSize = 0, fileType = FTDirectory, fileModTime = 1517265003}
FileInfo {filePath = "/home/xxxxxxxxxxxxxx/.stack/precompiled/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/ghc-8.2.2/2.0.1.0", fileUserId = 501, fileUserName = "xxxxxxxxxxxxxx", fileGroupId = 501, fileGroupName = "xxxxxxxxxxxxxx", fileMode = 509, fileSize = 0, fileType = FTDirectory, fileModTime = 1517285004}
FileInfo {filePath = "/home/xxxxxxxxxxxxxx/.stack/precompiled/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/ghc-8.2.2/2.0.1.0/cereal-0.5.4.0@sha256:8695c775923e3a185c20e574e4b8202752f7274d94df9b291340d976ea09a742,2767", fileUserId = 501, fileUserName = "xxxxxxxxxxxxxx", fileGroupId = 501, fileGroupName = "xxxxxxxxxxxxxx", fileMode = 509, fileSize = 0, fileType = FTDirectory, fileModTime = 1517265098}
FileInfo {filePath = "/home/xxxxxxxxxxxxxx/.stack/precompiled/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/ghc-8.2.2/2.0.1.0/cereal-0.5.4.0@sha256:8695c775923e3a185c20e574e4b8202752f7274d94df9b291340d976ea09a742,2767/XvBt2WClc9gS1tpA1pZNS9TmIn3m5Pipp2P08CAFPcA=", fileUserId = 501, fileUserName = "xxxxxxxxxxxxxx", fileGroupId = 501, fileGroupName = "xxxxxxxxxxxxxx", fileMode = 436, fileSize = 527, fileType = FTNormal, fileModTime = 1517283950}
[cache-s3][Info ][Wed, 31 Jan 2018 05:39:33 UTC]: Progress: 60%, speed: 7.2 MiB/s
[cache-s3][Info ][Wed, 31 Jan 2018 05:39:43 UTC]: Progress: 70%, speed: 7.4 MiB/s
[cache-s3][Info ][Wed, 31 Jan 2018 05:39:53 UTC]: Progress: 80%, speed: 7.0 MiB/s
[cache-s3][Info ][Wed, 31 Jan 2018 05:40:04 UTC]: Progress: 90%, speed: 6.9 MiB/s
[cache-s3][Info ][Wed, 31 Jan 2018 05:40:17 UTC]: Progress: 100%, speed: 5.6 MiB/s
[cache-s3][Info ][Wed, 31 Jan 2018 05:40:17 UTC]: <cache-s3/xxxxxxxxxxxxxxxxxxxxxxxx/master.lts-10.3.stack.cache> - Successfully restored previous cache
```

However, I have downloaded the cache file manually from S3 and ran:

```
tar -tvf master.lts-10.3.stack.cache
(...)
drwxrwxr-x  0 xxxxxxxxxxxxxx xxxxxxxxxxxxxx         0 30 Jan 09:30 /home/xxxxxxxxxxxxxx/.stack/precompiled
drwxrwxr-x  0 xxxxxxxxxxxxxx xxxxxxxxxxxxxx         0 30 Jan 09:30 /home/xxxxxxxxxxxxxx/.stack/precompiled/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743
drwxrwxr-x  0 xxxxxxxxxxxxxx xxxxxxxxxxxxxx         0 30 Jan 09:30 /home/xxxxxxxxxxxxxx/.stack/precompiled/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/ghc-8.2.2
drwxrwxr-x  0 xxxxxxxxxxxxxx xxxxxxxxxxxxxx         0 30 Jan 15:03 /home/xxxxxxxxxxxxxx/.stack/precompiled/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/ghc-8.2.2/2.0.1.0
drwxrwxr-x  0 xxxxxxxxxxxxxx xxxxxxxxxxxxxx         0 30 Jan 09:31 /home/xxxxxxxxxxxxxx/.stack/precompiled/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/ghc-8.2.2/2.0.1.0/cereal-0.5.4.0@sha256:8695c775923e3a185c20e574e4b8202752f7274d94df9b291340d976ea09a742,2767
-rw-rw-r--  0 xxxxxxxxxxxxxx xxxxxxxxxxxxxx       527 30 Jan 14:45 /home/xxxxxxxxxxxxxx/.stack/precompiled/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/ghc-8.2.2/2.0.1.0/cereal-0.5.4.0@sha256:8695c775923e3a185c20e574e4b8202752f7274d94df9b291340d976ea09a742,2767/XvBt2WClc9gS1tpA1pZNS9TmIn3m5Pipp2P08CAFPcA=
drwxrwxr-x  0 xxxxxxxxxxxxxx xxxxxxxxxxxxxx         0 30 Jan 14:53 /home/xxxxxxxxxxxxxx/.stack/precompiled/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/ghc-8.2.2/2.0.1.0/free-4.12.4@sha256:51279b76403ea6e900a02fbf2ab9c90eff7ba17c427f1d4caa98633c693b26db,3243
-rw-rw-r--  0 xxxxxxxxxxxxxx xxxxxxxxxxxxxx       515 30 Jan 14:53 /home/xxxxxxxxxxxxxx/.stack/precompiled/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/ghc-8.2.2/2.0.1.0/free-4.12.4@sha256:51279b76403ea6e900a02fbf2ab9c90eff7ba17c427f1d4caa98633c693b26db,3243/HpTjdNCHTBz1UBDIXua6crYg7ApZ5UDtJpbZMz2JgQQ=
drwxrwxr-x  0 xxxxxxxxxxxxxx xxxxxxxxxxxxxx         0 30 Jan 14:55 /home/xxxxxxxxxxxxxx/.stack/precompiled/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/ghc-8.2.2/2.0.1.0/aeson-casing-0.1.0.5@sha256:dfa52ef1ceaed3e22f5c157703359828aa6d4d295761480cf8a21fa7d3af153b,1466
(...)
```

There is a lot of files in `/home/xxxxxxxxxxxxxx/.stack/precompiled/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/ghc-8.2.2/2.0.1.0`, but only one is extracted by `cache-s3`.

As before, I have redacted some information but kept the number of characters the same - I'm seeing paths _dangerously_ close to 256 characters so it might be relevant.

**Comment by lehins** [comment]
@koterpillar You are absolutely correct on the fact that path length is the cause of the issue here. GNU tar support for long file names was recently broken in tar-conduit. I noticed the same thing last night and was able to track down and fix the problem. There is a new test version should be uploaded to the release soon: https://github.com/fpco/cache-s3/releases/tag/v0.1.1b

**Comment by koterpillar** [comment]
Thank you, v0.1.1b works for me!

**Comment by lehins** [comment]
Fixed in [v0.1.2](https://github.com/fpco/cache-s3/releases/tag/v0.1.2)
